### ISSUE TITLE
Ubuntu 20.04

### ISSUE THREAD

**Issue by colindaven** [ARTbio/GalaxyKickStart#370]
Hi,

we're attempting to use GKS on an Ubuntu 20.04 VM (our whole physical/virtual cluster has been upgraded to this now). 

Is there a timeline for an official update to 20.04 ? 

If we continue and are successful would you like us to share the packages required ?

Best wishes,
Colin

**Comment by drosofff** [comment]
Hi, @colindaven 
Thanks for using GKS. I'll open a 20.04 support branch asap. Of course you are welcome to contribute. It is very useful to have feedback from GKS users.
Kind regards
Chris

**Comment by colindaven** [comment]
Hi Chris,
excellent, thanks. It's going quite well so far.
As you probably expect, many of the changes will be going into the upstream repositories, which we've started to fork and modify.
Colin

**Comment by drosofff** [comment]
Hi @colindaven 
A quick update.
I have done a big refresh of galaxykickstart, including switching to github actions for CI, final switch to python 3 (python 2.7 is definitely dead for artbio now ;-), fixing the docker image artbio/galaxykickstart:18.04 (by now the only working clean container).
As usual, the work involved adaptation in the other ansible-galaxy-... repos.

Well I am now ready (and motivated) to switch galaxykickstart to ubuntu 20.04, especially be github action more easily allow parallel CI workflow with various ubuntu version (the ressources are better than with travis-ci). It should be quick hopefully.

What I can finally share with you is that I am really considering dropping the hooks with the others galaxyproject.ansible repo, and reimporting the artbio forks of these repos (as described in the requirements.txt file) as "flat" roles in galaxykickstart.

It's been for quite a long time a pain in the ass to get support from the community to have these repos active and responding.  I have a couple a merge requests lagging there for 1 or 2 years. I have also tried to poke people to get their views on the future of these ansible ressources, but without too much success. The net results from this situation is that I can only stand on my forks that are ahead or diverging. 
Thus, I am sad to say it, and it's a pity that potential issues that I may have missed were not transparently discussed, but maybe time has came to cut the umbilical cord with repos in a vegetative state...

In any case, galaxykickstart remains an important ansible playbook that we use in production in ARTbio, and I won't drop it out before a good while...

Regards
Chris

**Comment by colindaven** [comment]
Dear Chris,

thanks for this. We were successful in getting the GKS playbook to work in Ubuntu 20.04. This is likely a far less comprehensive install than what you use at ARTbio, but it is sufficient for us and our users.

We only pushed ahead on 20.04 as we had to. We upgraded the cluster from 16.04 to 20.04 and were suddenly at the bleeding edge. I hope some of our changes were useful to you, but as you say I don't think the PRs have been commented on or accepted at the community ansible repos. I guess this will change if sufficient demand is present, especially for cloud VM instances etc ? Or are some community members moving away from ansible now ? It's still great for our purposes. Concrete feedback from our side: we just need the basic playbook for installation on a VM, and that will remain for some time as far as I can see.

It is very good to know you plan to keep things going at your institute and maintain the playbooks. Thank you for this!

It may be interesting for us to try out the container versions in future. In the past we dallied with the various docker containers, but the VM approach seems easier for admin.

Best wishes,
Colin
